addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1308,221 +1308,276 @@ registry.background = SnippetOption.extend({
+        this.$overlayBackground = this.$overlayContent.find('.o_overlay_background');
+
+        this.$backgroundOverlay.find('.o_btn_apply').on('click', () => {
+            this.$target.css('background-position', this.$bgDragger.css('background-position'));
+            this._toggleOverlay(false);
+        });
+        this.$backgroundOverlay.find('.o_btn_discard').on('click', () => this._toggleOverlay(false));
+
+        $('#oe_manipulators').append(this.$backgroundOverlay);
     },
     /**
-     * Restores the target style before last edition made with the option.
+     * Sets the overlay in the right place so that the draggable background
+     * renders over the target, and size the background item like the target.
      *
      * @private
      */
-    _discardChanges: function () {
-        this._clean();
-        if (this.previousState) {
-            this.$target.addClass(this.previousState[0]).css({
-                'background-size': this.previousState[1],
-                'background-position': this.previousState[2],
+    _dimensionOverlay: function () {
+        const $wrapwrap = $('#wrapwrap');
+        const targetOffset = this.$target.offset();
+
+        this.$backgroundOverlay.css({
+            width: $wrapwrap.innerWidth(),
+            height: $wrapwrap.innerHeight(),
+            top: -$wrapwrap.innerHeight(),
+        });
+
+        this.$overlayContent.css({
+            cursor: 'grab',
+            top: targetOffset.top - $wrapwrap.offset().top,
+            left: targetOffset.left - $wrapwrap.offset().left,
+        });
+
+        if (this.$bgDragger) {
+            this.$bgDragger.css({
+                width: `${this.$target.innerWidth()}px`,
+                height: `${this.$target.innerHeight()}px`,
             });
         }
     },
     /**
-     * Updates the visual representation of the chosen background position.
+     * Toggles the overlay's display and renders a background clone inside of it.
      *
      * @private
+     * @param {boolean} activate toggle the overlay on (true) or off (false)
      */
-    _updatePosInformation: function () {
-        this.modal.$('.o_bg_img_opt_ui_info .o_x').text(this.bgPos.x);
-        this.modal.$('.o_bg_img_opt_ui_info .o_y').text(this.bgPos.y);
-        this.$focus.css({
-            left: this.bgPos.x,
-            top: this.bgPos.y,
+    _toggleOverlay: function (activate) {
+        // Check if overlay is already in requested state.
+        if (this.$backgroundOverlay.is('.oe_active') === activate) {
+            return;
+        }
+        if (!activate) {
+            this.$backgroundOverlay.removeClass('oe_active');
+            this.$manipulators.removeClass('d-none');
+            this.trigger_up('cover_update');
+
+            $(document).off('click', this._onDocumentClicked);
+            return;
+        }
+        
+        this.$manipulators = $('#oe_manipulators > :not(.o_background_position_overlay)').addClass('d-none');
```

Comment:
Same problem as with deactivating a snippet while the carousel is sliding, the menu would show the "Blocks" section of the menu instead of staying on the snippet being currently modified which looks weird. Ideally this would indeed be handled by the snippetMenu, but with a different event to just hide the overlay of the current snippet, and another one to show it again. It just seemed like a lot of infrastructure work for something that may only be used by one snippet, but if you think it's worth it to add that feature to the snippetMenu directly, I will do that.

Reply:
Mmh, maybe not, I'll think about it when a second case appears. Meanwhile, you can still replace by `this.$overlay.siblings(':not(.o_background_position_overlay)')` to find those elements.